File: 733-flood-fill/733-flood-fill.py
class Solution(object):
    def floodFill(self, image, sr, sc, newColor):
        """
        :type image: List[List[int]]
        :type sr: int
        :type sc: int
        :type color: int
        :rtype: List[List[int]]
        """
        color = image[sr][sc]
        if color == newColor: return image
        
        def dfs(i, j):
            if image[i][j] == color:
                image[i][j] = newColor
                if i > 0: dfs(i-1, j)
                if i < len(image)-1: dfs(i+1, j)
                if j > 0: dfs(i, j-1)
                if j < len(image[0])-1: dfs(i, j+1)
                    
        dfs(sr, sc)
        return image
        
        

File: 733-flood-fill/733-flood-fill.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        int color = image[sr][sc];
        if (color != newColor) dfs(image, sr, sc, color, newColor);
        return image;
    }
    
    public int[][] dfs(int[][] image, int i, int j, int color, int newColor) {
        //if (image[i][j] == color) {
            //image[i][j] = newColor;
            //if (i > 0) dfs(image, i-1, j, color, newColor);
           // if (i < image.length-1) dfs(image, i+1, j, color, newColor);
          //  if (j > 0) dfs(image, i, j-1, color, newColor);
          //  if (j < image[0].length-1) dfs(image, i, j+1, color, newColor);
      //  }
        
       if (image[i][j] != color) return image; 
       image[i][j] = newColor;
       if (i > 0) dfs(image, i-1, j, color, newColor);
       if (i < image.length-1) dfs(image, i+1, j, color, newColor);
       if (j > 0) dfs(image, i, j-1, color, newColor);
       if (j < image[0].length-1) dfs(image, i, j+1, color, newColor);
       return image;
    }
}